src/app/public/modules/wait/wait-adapter.service.ts

```diff
@@ -21,13 +36,69 @@ export class SkyWaitAdapterService {
     waitEl: ElementRef,
     isFullPage: boolean,
     isWaiting: boolean,
+    id: string,
     isNonBlocking = false
-  ): void {
+  ) {
     let busyEl = isFullPage ? document.body : waitEl.nativeElement.parentElement;
     let state = isWaiting ? 'true' : undefined;
 
     if (!isNonBlocking) {
       this.renderer.setElementAttribute(busyEl, 'aria-busy', state);
+
+      if (isWaiting) {
+        // Propagate tab navigation if attempted into waited element
+        if (!isFullPage) {
+          let focusListenerFunc = this.renderer.listen(this.windowRef.getWindow(), 'keyup', (event: any) => {
+            if (event.key.toLowerCase() === 'tab' && busyEl.contains(document.activeElement)) {
+              this.focusNextElement(busyEl, event.shiftKey);
+            }
+          });
+          this.parentListeners[id] = focusListenerFunc;
+        } else {
+          // Prevent tab navigation within the waited page
+          let endListenerFunc = this.renderer.listen(busyEl, 'keydown', (event: KeyboardEvent) => {
+            if (event.key.toLowerCase() === 'tab') {
+              event.preventDefault();
+            }
+          });
+          this.parentListeners[id] = endListenerFunc;
+        }
+      } else if (id in this.parentListeners) {
+        // Clean up existing listener
+        this.parentListeners[id]();
+        delete this.parentListeners[id];
+      }
+    }
+  }
+
+  private focusNextElement(parentElement: any, shiftKey: boolean): void {
+    // Select all possible focussable elements
+    let focussableElements =
+      'a[href], ' +
+      'area[href], ' +
+      'input:not([disabled]):not([tabindex=\'-1\']), ' +
+      'button:not([disabled]):not([tabindex=\'-1\']), ' +
+      'select:not([disabled]):not([tabindex=\'-1\']), ' +
+      'textarea:not([disabled]):not([tabindex=\'-1\']), ' +
+      'iframe, object, embed, ' +
+      '*[tabindex]:not([tabindex=\'-1\']), ' +
```

Comment:
Unfortunately these are "or"s. to if I have a selector that says: `input, *[tabindex]:not([*tabindex=\'-1\'])` it will select the element: `<input tabindex="-1" />` because it does match that first part of the selector.

To be sure, I just checked this and confirmed it functions that way